www/js/app.js

```diff
@@ -1325,83 +1399,117 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
 
-        var iframeArticleContent = document.getElementById('articleContent');
-        
-        iframeArticleContent.onload = function() {
-            iframeArticleContent.onload = function(){};
+        // Code below will run after we have written the new article to the articleContainer
+        var windowLoaded = function() {
             $("#articleList").empty();
             $('#articleListHeaderMessage').empty();
             $('#articleListWithHeader').hide();
             $("#prefix").val("");
-            
-            var iframeContentDocument = iframeArticleContent.contentDocument;
-            if (!iframeContentDocument && window.location.protocol === 'file:') {
-                alert("You seem to be opening kiwix-js with the file:// protocol, which is blocked by your browser for security reasons."
-                        + "\nThe easiest way to run it is to download and run it as a browser extension (from the vendor store)."
-                        + "\nElse you can open it through a web server : either through a local one (http://localhost/...) or through a remote one (but you need SSL : https://webserver/...)"
-                        + "\nAnother option is to force your browser to accept that (but you'll open a security breach) : on Chrome, you can start it with --allow-file-access-from-files command-line argument; on Firefox, you can set privacy.file_unique_origin to false in about:config");
-                return;
+            if (appstate.target === 'iframe' && !articleContainer.contentDocument && window.location.protocol === 'file:') {
+                if (!appstate.fileAccessWarning) {
+                    uiUtil.systemAlert("<p>You seem to be opening kiwix-js with the file:// protocol, which blocks access to the app's iframe. "
+                    + "We have tried to open your article in a separate window. You may be able to use it with limited functionality.</p>"
+                    + "<p>The easiest way to run this app fully is to download and run it as a browser extension (from the vendor store). "
+                    + "Alternatively, you can open it through a web server: either use a local one (http://localhost/...) "
+                    + "or a remote one. For example, you can try your ZIM out right now with our online version of the app: "
+                    + "<a href='https://kiwix.github.io/kiwix-js/'>https://kiwix.github.io/kiwix-js/</a>.</p>"
+                    + "<p>Another option is to force your browser to accept file access (a potential security breach): "
+                    + "on Chrome, you can start it with the <code>--allow-file-access-from-files</code> command-line argument; on Firefox, "
+                    + "you can set <code>privacy.file_unique_origin</code> to <code>false</code> in about:config.</p>");
+                }
+                articleContainer = window.open('', dirEntry.title, 'toolbar=0,location=0,menubar=0,width=800,height=600,resizable=1,scrollbars=1');
```

Comment:
This opens as about:blank, which is not considered cross-origin, and so is not blocked.